src/main/java/org/betonquest/betonquest/api/CountingObjective.java

```diff
@@ -79,10 +80,10 @@ public final CountingData getCountingData(final Profile profile) {
      * {@link #defaultNotifyMessageName} if set, otherwise no notification will be sent, even if {@link #notify} is
      * {@code true}.
      *
-     * @param profile the {@link Profile} to act for
+     * @param profile the {@link OnlineProfile} to act for
      * @return {@code true} if the objective is completed; {@code false} otherwise
      */
-    protected final boolean completeIfDoneOrNotify(final Profile profile) {
+    protected final boolean completeIfDoneOrNotify(final OnlineProfile profile) {
```

Comment:
This is not correct, this still need to be a Profile, because objectives can still be completed for offline players, for example with a command. I think you need to check your PR completely before i finish my review